src/store/index.js

```diff
@@ -0,0 +1,53 @@
+import { createStore } from 'vuex'
+
+export default createStore({
+  state: {
+    users: [],
+    error: '',
+    isLoading: false,
+  },
+
+  mutations: {
+    setUsers(state, users){
+      state.users = users;
+      localStorage.setItem("users", JSON.stringify(state.users));
+    },
+    loading (state,  loadingStatus){
+      state.isLoading = loadingStatus
+    },
+    postError(state, err) {
+      state.error = err;
+    }
+  },
+
+  actions: {
+    async fetchUsers({commit}) {
+      const savedUsers = localStorage.getItem("users");
+      commit("loading", true);
+      
+      if (savedUsers !== null) {
+        commit("setUsers", JSON.parse(savedUsers));        
+      } else {
+        try {        
+          const response = await fetch("https://jsonplaceholder.typicode.com/users");
+          const data = await response.json();
```

Comment:
when you activate any checkboxes on users, you can reload page. After page reload will you still have updated users? Or you will have the value from response?